### ISSUE TITLE
inferences detection control option ?

### ISSUE THREAD

**Issue by Nal44** [UFISH-Team/U-FISH#5]
Hi, 

I really life your plugin and it is doing a pretty good job overall.
But it is too sensitive and enhancing spots that are close to background or too detecting multiples small spots in 3d when in the diffractions rings.

Is there a way to control the detection sensitivity before enhancing ? (min/ max or a cut off/ thresholding ) ? 

Thanks
antho

**Comment by Nanguage** [comment]
Hi Antho,

It's great to see that you have tried U-FISH and provided feedback. Regarding this issue, my suggestion is to try setting a higher intensity_threshold parameter. Alternatively, you can try denoising the original image, etc. This may depend on the specific situation. If possible, you can provide a sample data for us to give more specific advice.

Best,
Weize

**Comment by Nal44** [comment]
Hi Weize,

Thanks,
I played with the p threshold using the plugin in Napari, but this only affect the spots ?
Using inferences, how can I adjust the intensity_threshold parameter (in napari OR using a notebook) ?

I will send you 2 images (one 2D and one 3D) to your email.

The main problem in 2D : it detects " too much" , need a way to control the detection before the enhancement (hence false positive).

The 3D problem : my main problem is to remove the Halo from the diffraction rings, Ufish is doing a great job overall BUT it find "spots in the halos rings" hence introducing false positives spots, hence if this can be controlled somehow (reducing /removing false positive) then it will highly suitable for my needs.

THanks a lot,
antho

**Comment by Nanguage** [comment]
Yes, it only affects spots. If you want to affect enhancement, there are only two ways:

1. Modify the input image, such as adjusting contrast, denoising, etc.
2. Fine-tune the network, which may be the most effective method but requires some training data (not necessarily a lot; we have tested that around 30 images of size 512*512 pixels can achieve good results). You can refer to our [colab notebook](https://colab.research.google.com/github/UFISH-Team/U-FISH/blob/main/notebooks/ufish_finetune.ipynb) and perform fine-tuning training on it.

Other suggestions will be given to you after we have seen your images. Thank you again for using our tool!

**Comment by Nal44** [comment]
Ok,
 Fine tuning might be the way to go, but making the ground truth is time consuming...

Did you receive an invite from "on demand file transfer " ?

Thanks
antho

**Comment by Nanguage** [comment]
Yes, I have received it. You can establish the ground truth based on the predictions made by U-FISH, which will not take too much time. In fact, we also used a similar pre-labeling method when establishing the initial dataset.

**Comment by Nal44** [comment]
Ok, that was indeed my plan, 

One thing though, once enhanced the intensities are changed, is there a way to keep the raw intensities after detection ?

One way is to use Ufish for the detection, then use the layer as a mask and record the intensities on the raw image, but if there is a more direct way that will be better (possible improvement ?)

**Comment by Nanguage** [comment]
Yes, this is the result caused by the working principle of ufish. The network of ufish will map the original signal strength to a range close to 0-1. Currently, there is no good solution. The method you mentioned using masks is a feasible approach. Thank you very much for your suggestion. We will incorporate this issue into our future improvement direction.

**Comment by Nal44** [comment]
Hi, 

I made a dataset in 3d for fine tuning(tif +csv) , hence I have 3 axes.
The notebook is mainly for 2d, so how do I refine using the 3 axes ? 

Thanks a lot,

**Comment by Nanguage** [comment]
Hi,

U-FISH 3d is based on 2d network and adopts a strategy to apply to 3d data. So my suggestion is to first divide the 3d data into 2d data along the z-axis, and then use the method in Notebook to finetune. See how the effect is. If there are still problems, you can divide the 3D data into a series of 2D data along x, y and z for finetune.

**Comment by Nal44** [comment]
Ok, I assumed that but wanted to double check :D.

THanks a lot
